api/nodes/permissions.py

```diff
@@ -63,6 +65,36 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+
+class ContributorOrPublicForComments(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, Comment, CommentReport)), 'obj must be a Node, Comment or CommentReport, got {}'.format(obj)
+        auth = get_user_auth(request)
+        if isinstance(obj, CommentReport):
+            comment_id = request.parser_context['kwargs']['comment_id']
+            comment = Comment.load(comment_id)
+            if not comment:
+                raise NotFound
+            node = comment.node
+        elif isinstance(obj, Comment):
+            comment = obj
+            node = obj.node
+        elif isinstance(obj, Node):
+            node = obj
+
+        view_name = view.get_view_name()
+        is_report_view = (view_name == 'Comment Reports List' or view_name == 'Comment Report Detail')
```

Comment:
Although we selectively allow scoped imports, I'm more generally bothered by the fact that we want a single permissions class to do different things based on view. The problem we're trying to solve is that different views need different permissions behavior, and DRF has a pretty well defined way to do that.

Rather than keep hacking around that, why not be honest and make these separate permissions classes? We could subclass `ContributorOrPublicForComments`, and override the specific different in a view-specific helper method.